File: app/src/main/java/com/ivanmagda/habito/analytics/HabitoAnalytics.java
package com.ivanmagda.habito.analytics;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;

import com.google.firebase.analytics.FirebaseAnalytics;
import com.google.firebase.auth.FirebaseUser;
import com.ivanmagda.habito.models.Habit;

public final class HabitoAnalytics {

    private static FirebaseAnalytics sFirebaseAnalytics;

    public static void initAnalytics(Context context) {
        // Obtain the FirebaseAnalytics instance.
        sFirebaseAnalytics = FirebaseAnalytics.getInstance(context);
    }

    public static FirebaseAnalytics getInstance(Context context) {
        if (sFirebaseAnalytics == null) initAnalytics(context);
        return sFirebaseAnalytics;
    }

    public static void logCreateHabitWithName(@NonNull final String name) {
        Bundle params = new Bundle();
        params.putString("habit_name", name);
        sFirebaseAnalytics.logEvent("create_habit", params);
    }

    public static void logAppOpen() {
        Bundle bundle = new Bundle();
        bundle.putLong(FirebaseAnalytics.Param.START_DATE, System.currentTimeMillis());
        sFirebaseAnalytics.logEvent(FirebaseAnalytics.Event.APP_OPEN, bundle);
    }

    public static void logLogin(FirebaseUser user) {
        Bundle bundle = new Bundle();
        bundle.putLong(FirebaseAnalytics.Param.START_DATE, System.currentTimeMillis());
        bundle.putString(FirebaseAnalytics.Param.ITEM_ID, user.getUid());
        bundle.putString(FirebaseAnalytics.Param.ITEM_NAME, user.getDisplayName());
        sFirebaseAnalytics.logEvent(FirebaseAnalytics.Event.LOGIN, bundle);
    }

    public static void logViewHabitListItem(Habit habit) {
        Bundle bundle = new Bundle();
        bundle.putString(FirebaseAnalytics.Param.ITEM_ID, habit.getId());
        bundle.putString(FirebaseAnalytics.Param.ITEM_NAME, habit.getRecord().getName());
        bundle.putInt(FirebaseAnalytics.Param.SCORE, habit.getRecord().getScore());
        bundle.putInt("color", habit.getRecord().getColor());
        sFirebaseAnalytics.logEvent(FirebaseAnalytics.Event.VIEW_ITEM_LIST, bundle);
    }

    private HabitoAnalytics() {
    }

}

File: app/src/main/java/com/ivanmagda/habito/analytics/HabitoAnalytics.kt
package com.ivanmagda.habito.analytics

import android.content.Context
import android.os.Bundle

import com.google.firebase.analytics.FirebaseAnalytics
import com.google.firebase.auth.FirebaseUser
import com.ivanmagda.habito.models.Habit

object HabitoAnalytics {

    private lateinit var firebaseAnalytics: FirebaseAnalytics

    fun initAnalytics(context: Context) {
        firebaseAnalytics = FirebaseAnalytics.getInstance(context)
    }

    fun logCreateHabitWithName(name: String) {
        val params = Bundle()
        params.putString("habit_name", name)

        firebaseAnalytics.logEvent("create_habit", params)
    }

    fun logAppOpen() {
        val bundle = Bundle()
        bundle.putLong(FirebaseAnalytics.Param.START_DATE, System.currentTimeMillis())

        firebaseAnalytics.logEvent(FirebaseAnalytics.Event.APP_OPEN, bundle)
    }

    fun logLogin(user: FirebaseUser) {
        val bundle = Bundle()
        bundle.putLong(FirebaseAnalytics.Param.START_DATE, System.currentTimeMillis())
        bundle.putString(FirebaseAnalytics.Param.ITEM_ID, user.uid)
        bundle.putString(FirebaseAnalytics.Param.ITEM_NAME, user.displayName)

        firebaseAnalytics.logEvent(FirebaseAnalytics.Event.LOGIN, bundle)
    }

    fun logViewHabitListItem(habit: Habit) {
        val bundle = Bundle()
        bundle.putString(FirebaseAnalytics.Param.ITEM_ID, habit.id)
        bundle.putString(FirebaseAnalytics.Param.ITEM_NAME, habit.record.name)
        bundle.putInt(FirebaseAnalytics.Param.SCORE, habit.record.score)
        bundle.putInt("color", habit.record.color)

        firebaseAnalytics.logEvent(FirebaseAnalytics.Event.VIEW_ITEM_LIST, bundle)
    }
}
